src/clf/morpho/morphoDashboardComputer.js

```diff
@@ -0,0 +1,333 @@
+            if (riskData.riskLevel > vaultData.riskLevel) {
+                vaultData.riskLevel = riskData.riskLevel;
+            }
+            pairData['volatility'] = riskData.volatility;
+            pairData['liquidity'] = riskData.liquidity;
+            vaultData.subMarkets.push(pairData);
+        }
+    }
+
+    return vaultData;
+}
+
+function getLiquidationBonusForLtv(ltv) {
+    switch (ltv) {
+        default:
+            throw new Error(`No liquidation bonus for ltv ${ltv}`);
+        case 0.98:
+            return 50;
+        case 0.965:
+            return 100;
+        case 0.945:
+            return 150;
+        case 0.915:
+            return 250;
+        case 0.86:
+            return 400;
+        case 0.77:
+            return 700;
+        case 0.625:
+            return 1250;
+    }
+}
+
+async function getVaultMarkets(vault, currentBlock) {
+    try {
+        const marketIds = [];
+        const withdrawQueueLengthBn = await vault.withdrawQueueLength({ blockTag: currentBlock });
+        const vaultQueueLength = Number(withdrawQueueLengthBn.toString());
+        for (let i = 0; i < vaultQueueLength; i++) {
+            const marketId = await vault.withdrawQueue(i, { blockTag: currentBlock });
+            marketIds.push(marketId);
+        }
+
+        return marketIds;
+    } catch (e) {
+        console.warn(e);
+        return [];
+    }
+}
+
+async function getHistoricalPrice(tokenAddress, dateUnixSec) {
+    const apiUrl = `https://coins.llama.fi/prices/historical/${dateUnixSec}/ethereum:${tokenAddress}?searchWidth=12h`;
+    const historicalPriceResponse = await retry(axios.get, [apiUrl], 0, 100);
+    return historicalPriceResponse.data.coins[`ethereum:${tokenAddress}`].price;
+}
+
+function findRiskLevelFromParameters(volatility, liquidity, liquidationBonus, ltv, borrowCap) {
+    const sigma = volatility;
+    const d = borrowCap;
+    const beta = liquidationBonus;
+    const l = liquidity;
+    ltv = Number(ltv) / 100;
+
+    const sigmaTimesSqrtOfD = sigma * Math.sqrt(d);
+    const ltvPlusBeta = ltv + beta;
+    const lnOneDividedByLtvPlusBeta = Math.log(1 / ltvPlusBeta);
+    const lnOneDividedByLtvPlusBetaTimesSqrtOfL = lnOneDividedByLtvPlusBeta * Math.sqrt(l);
+    const r = sigmaTimesSqrtOfD / lnOneDividedByLtvPlusBetaTimesSqrtOfL;
+
+    return r;
+}
+
+function recordResults(results) {
+    if (!fs.existsSync(`${DATA_DIR}/precomputed/morpho-dashboard/`)) {
+        fs.mkdirSync(`${DATA_DIR}/precomputed/morpho-dashboard/`, { recursive: true });
+    }
+    const summaryFilePath = path.join(DATA_DIR, 'precomputed/morpho-dashboard/morpho-summary.json');
+    const objectToWrite = JSON.stringify(results, null, 2);
+    console.log('recording results');
+    try {
+        fs.writeFileSync(summaryFilePath, objectToWrite, 'utf8');
+    }
+    catch (error) {
+        console.error(error);
+        console.log('Morpho Computer failed to write files');
+    }
```

Comment:
Don't do this try/catch, you need to throw so that the monitoring will catch the error and send the error to the monitoring platform